File: site/public_html/staging-scripts/load_milestone_definitions.php
<?php
/**
 * @deprecated Superseded by site/public_html/ops/run_prepare.php seed-catalog (REP-014).
 *
 * Work / Laragon dev:
 *   php site/public_html/ops/run_prepare.php seed-catalog --target local-work
 *   php site/public_html/ops/run_prepare.php seed-catalog --target local-dev
 *
 * Staging (until ops + repo data/ are on server): this file reloads from
 * public_html/staging-data/milestones_definitions_seed.json (WinSCP copy of seed).
 *
 * Historical CLI after staging-sql/010:
 *   php staging-scripts/load_milestone_definitions.php
 */
declare(strict_types=1);

require_once __DIR__ . '/_staging_milestones_bootstrap.php';

$con = k2_staging_milestones_bootstrap();

if (!k2_staging_table_exists($con, 'milestone_definitions')) {
    fwrite(STDERR, "Apply staging-sql/010_milestone_definitions.sql first.\n");
    exit(1);
}

$seedPath = dirname(__DIR__) . '/staging-data/milestones_definitions_seed.json';
if (!is_file($seedPath)) {
    fwrite(STDERR, "Missing seed: {$seedPath}\n");
    exit(1);
}

$payload = json_decode(file_get_contents($seedPath), true, 512, JSON_THROW_ON_ERROR);
$rows = $payload['definitions'] ?? null;
if (!is_array($rows)) {
    fwrite(STDERR, "Invalid seed: missing definitions array.\n");
    exit(1);
}

$tierMap = [
    'aspirational' => 'aspirational',
    'dedicated' => 'veteran',
    'accomplished' => 'key',
    'legendary' => 'legendary',
];

$dbRes = $con->query('SELECT DATABASE() AS db');
$dbName = ($dbRes && ($r = $dbRes->fetch_assoc())) ? (string) $r['db'] : '?';
if ($dbRes) {
    $dbRes->free();
}
fwrite(STDERR, "Note: prefer ops/run_prepare.php seed-catalog when available.\n");
echo "REP-014: loading milestone_definitions on {$dbName}...\n";
$con->query('TRUNCATE TABLE milestone_definitions');

$stmt = $con->prepare(
    'INSERT INTO milestone_definitions (
        milestone_key, display_name, tier_band, chart_token,
        rule_short, description, sort_order, icon
    ) VALUES (?, ?, ?, ?, ?, NULL, ?, NULL)'
);
if ($stmt === false) {
    fwrite(STDERR, 'Prepare failed: ' . $con->error . PHP_EOL);
    exit(1);
}

foreach ($rows as $i => $row) {
    $seedTier = (string) ($row['tier_band'] ?? '');
    $tier = $tierMap[$seedTier] ?? null;
    if ($tier === null) {
        fwrite(STDERR, "Unknown tier_band in seed: {$seedTier}\n");
        exit(1);
    }
    $sort = $i + 1;
    $stmt->bind_param(
        'sssssi',
        $row['milestone_key'],
        $row['display_name'],
        $tier,
        $row['chart_token'],
        $row['rule_short'],
        $sort
    );
    if (!$stmt->execute()) {
        fwrite(STDERR, 'Insert failed: ' . $stmt->error . PHP_EOL);
        exit(1);
    }
}
$stmt->close();

$res = $con->query('SELECT COUNT(*) AS n FROM milestone_definitions');
$row = $res ? $res->fetch_assoc() : null;
if ($res) {
    $res->free();
}
$n = (int) ($row['n'] ?? 0);
echo "Catalog rows: {$n}\n";
if ($n !== count($rows)) {
    fwrite(STDERR, "Expected " . count($rows) . " rows, got {$n}\n");
    exit(1);
}

echo "Done.\n";
mysqli_close($con);

File: scripts/oneoff/load_milestone_definitions.py
#!/usr/bin/env python3
"""
Load milestone_definitions from ops/data/milestones_definitions_seed.json (local/staging).

Usage (repo root):
  python scripts/oneoff/load_milestone_definitions.py
"""
from __future__ import annotations

import json
import logging
import sys
from pathlib import Path

_REPO = Path(__file__).resolve().parents[2]
if str(_REPO) not in sys.path:
    sys.path.insert(0, str(_REPO))

from scripts.ladder.config import load_db_config  # noqa: E402
from scripts.ladder.engine import connect  # noqa: E402

log = logging.getLogger("load_milestone_definitions")
SEED = _REPO / "site" / "public_html" / "ops" / "data" / "milestones_definitions_seed.json"

# Seed tier labels (Phase 2) → product tier_band (Phase 3 / milestones-product-spec.md)
TIER_BAND_PRODUCT = {
    "aspirational": "aspirational",
    "dedicated": "veteran",
    "accomplished": "key",
    "legendary": "legendary",
}


def main() -> None:
    logging.basicConfig(level=logging.INFO, format="%(levelname)s %(message)s")
    payload = json.loads(SEED.read_text(encoding="utf-8"))
    rows = payload["definitions"]
    con = connect(load_db_config(), dry_run=False)
    cur = con.cursor()
    cur.execute("TRUNCATE TABLE milestone_definitions")
    sql = """
        INSERT INTO milestone_definitions (
            milestone_key, display_name, tier_band, chart_token,
            rule_short, description, sort_order, icon
        ) VALUES (%s, %s, %s, %s, %s, NULL, %s, NULL)
    """
    for i, row in enumerate(rows, start=1):
        seed_tier = str(row["tier_band"])
        tier = TIER_BAND_PRODUCT.get(seed_tier)
        if tier is None:
            raise ValueError(f"Unknown tier_band in seed: {seed_tier!r} ({row['milestone_key']})")
        cur.execute(
            sql,
            (
                row["milestone_key"],
                row["display_name"],
                tier,
                row["chart_token"],
                row["rule_short"],
                i,
            ),
        )
    con.commit()
    cur.execute("SELECT COUNT(*) AS n FROM milestone_definitions")
    n = int(cur.fetchone()["n"])
    cur.close()
    con.close()
    if n != len(rows):
        raise SystemExit(f"Expected {len(rows)} rows, got {n}")
    log.info("Loaded %d milestone_definitions from %s", n, SEED.name)


if __name__ == "__main__":
    main()
